<launch>

    <!-- General Parameters -->
    <arg name="record_bag"      default="true" />
    <arg name="bag_path"      default="$(find coll_avoid)/bag/" />

    <!-- Drone WiFi UDP -->
    <param name="drone_ip"	type="string"	value="192.168.0.1" />
    <param name="drone_port"	type="int"	value="8889" />
    <param name="drone_timeout"	type="double"	value="5.0" />

    <!-- Sensor Bluetooth RFCOMM -->
    <param name="bt_port"		type="int"	value="3" />
    <param name="sensor_timeout"	type="double"	value="10.0" />

    <!-- Estimation Parameters -->
    <param name="use_drone_state"	type="bool"	value="false" />


    <node name="main" pkg="coll_avoid" type="main_node.py" output="screen" />
    <node name="drone" pkg="coll_avoid" type="drone_node.py" output="screen" />
    <node name="sensor" pkg="coll_avoid" type="sensor_node.py" output="screen" />

    <group if="$(arg record_bag)">
    	<node pkg="rosbag" type="record" name="record" args="-o $(arg bag_path) /cmd_vel /cmd_sensor /cmd_action /telemetry /sensor_meas /est" />
    </group>

</launch>
